File: Code/new_fitervari_backend/src/main/java/at/htl/model/Workout2Exercise.java
package at.htl.model;

import javax.persistence.*;

@Entity
@Table(name = "workout2exercise")
public class Workout2Exercise {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    private Workout workout;

    @ManyToOne(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    private Exercise exercise;
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/Workout2Exercise.kt
package at.htl_leonding.model

import javax.persistence.*

@Entity
@Table(name = "workout2exercise")
class Workout2Exercise(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @ManyToOne(cascade = [CascadeType.MERGE])
    var workout: Workout,
    @ManyToOne(cascade = [CascadeType.MERGE])
    var exerciseId: Exercise
)